src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -100,7 +102,14 @@
                         ;; Try to get a connection to make sure the db is ready
                         (.close (.getConnection datasource))
                         (when-let [{:keys [migration-db migration-dir replication-mode]} migration-opts]
-                          (when-not (= replication-mode :replica)
+                          ;; If we're a replica then pglogical will be
+                          ;; replicating our migrations for us, so we poll until
+                          ;; the migrations have been replicated
+                          (if (= replication-mode :replica)
+                            (loop []
```

Comment:
Not a big fan of nested `(loop [] (recur))`, TBH.  Can we move this branch of the if into a function?